LLDBDebugger/LLDBProtocol/LLDBConnector.cpp

```diff
@@ -438,7 +440,9 @@ void LLDBConnector::OnProcessOutput(clProcessEvent& event)
 void LLDBConnector::OnProcessTerminated(clProcessEvent& event)
 {
     wxDELETE(m_process);
-    Cleanup();
+
+    clDebugEvent e2(wxEVT_LLDB_CRASHED);
+    ProcessEvent(e2);
```

Comment:
Are we sure that `wxEVT_LLDB_CRASHED` is the correct one to send here? This will trigger a message to the user saying that LLDB has crashed. We also have `wxEVT_LLDB_EXITED` (`OnProcessTerminated` can be called also if the user kills LLDB and not crashed)

Also, I would prefer to fire the event using `AddPendingEvent` and not `ProcessEvent` (the `AddPendingEvent` schedules the event in an async mode unlike `ProcessEvent` which processes the event in sync mode)

Reply:
Yeah like I say I did wonder if it was the wrong event, but also thought a dialog box popping up would be useful info so I just hacked it to use wxEVT_LLDB_CRASHED. I can add another event, or I guess some user data can be attached to the event (e.g. "lldb" or "codelite-lldb", as appropriate) that the dialog box can use?

Sync/async - OK. The idiosyncrasies of wx events aren't clear to me yet :smile: I just did sync so it was effectively the same as the existing call to Cleanup() (as I understood it).

Cheers!

Reply:
You can use `wxEVT_LLDB_EXITED` (this sounds about right to me)

Reply:
So no message box you mean, since LLDBPlugin already binds wxEVT_LLDB_EXITED?